dynosaur_derive/src/where_clauses.rs

```diff
@@ -0,0 +1,41 @@
+use syn::visit_mut::VisitMut;
+use syn::{PredicateType, Signature, Type, TypeParamBound, WhereClause, WherePredicate};
+
+pub fn has_where_self_sized(sig: &mut Signature) -> bool {
+    let mut visitor = SelfSized(false);
+    visitor.visit_signature_mut(sig);
+    visitor.0
+}
+
+struct SelfSized(bool);
+
+impl VisitMut for SelfSized {
+    fn visit_where_clause_mut(&mut self, clause: &mut WhereClause) {
+        self.0 |= clause
+            .predicates
+            .iter()
+            .find(|predicate| match predicate {
+                WherePredicate::Type(PredicateType {
+                    bounded_ty: Type::Path(type_path),
+                    bounds,
+                    ..
+                }) => {
+                    type_path.path.segments[0].ident == "Self"
```

Comment:
Maybe just take the whole path as a string and compare it, that way we don't accidentally match against `where Self::AssocType: Sized`